Cassette/CASQueueFile.m

```diff
@@ -107,12 +107,29 @@ + (void)safelySetUpFile:(NSString *)path forManager:(NSFileManager *)fileManager
     NSError *tapeError;
 
     // Use a temporary file so we don't leave a partially-initialized file.
-    NSString *tempPath = [NSString stringWithFormat:@"%@.tmp", path];
+    NSString *tempPath = [path stringByAppendingPathExtension:@"tmp"];
 
     // Write the initial set of data for the file
     NSMutableData *fileBuffer = [NSMutableData dataWithLength:QueueFileInitialLength];
     writeInt(fileBuffer, 0, QueueFileInitialLength);
 
+    BOOL isDirectory = YES;
+
+    NSString *folderPath = [path stringByDeletingLastPathComponent];
+
+    BOOL doesFolderExistAlready = [fileManager fileExistsAtPath:folderPath
+                                                    isDirectory:&isDirectory];
+
+    if (!doesFolderExistAlready) {
+        tapeError = [CASError createError:CASErrorFileInitialization];
+        CASLOG(@"Creation of intermediary directories is not support, please create folder path: %@", folderPath);
+    }
+
+    if (doesFolderExistAlready && !isDirectory) {
```

Comment:
else if?

Reply:
actually the first check `doesFolderExistAlready` is not necessary..., I think this is cleaner/more readible as just 

`    if (!isDirectory) {` changing it to this.